### PR TITLE
Suppliers engine

### PR DESCRIPTION
**Pull request by sauloperez**
## WAT

Introduciendo el Data Model referente a suppliers.

### CODE REVIEW

**Comment by andresgutgon** [comment]
OMG katuma is on fire :fire: !

**Comment by enricostano** [comment]
OMG OMG OMG :scream_cat:

**Comment by sauloperez** [comment]
Hahahahaha, teneis un trigger en el móbil o qué? :trollface:

**Comment by enricostano** [comment]
Run the migrations and show me the schema! :stuck_out_tongue_winking_eye:

File: db/schema.rb [hunk 1]
```diff
@@ -99,10 +122,20 @@
...
+  end
+
+  add_index "suppliers", ["group_id"], name: "index_suppliers_on_group_id", using: :btree
+  add_index "suppliers", ["producer_id"], name: "index_suppliers_on_producer_id", using: :btree
+
   create_table "users", force: true do |t|
     t.string   "email",           null: false
-    t.string   "first_name",      null: false
-    t.string   "last_name",       null: false
+    t.string   "first_name"
```
Comment (sauloperez) [inline db/schema.rb]: Umm, alguien modificó la migración? en cualquier caso, no deberíamos obligar a tener nombre y apellido?

Comment (andresgutgon) [inline db/schema.rb]: si deberiamos

Comment (enricostano) [inline db/schema.rb]: I'm not sure, but if you think so let's go with required `first_name` and `last_name` attributes

File: db/schema.rb [hunk 2]
```diff
@@ -73,6 +73,29 @@
 
   add_index "oauth_applications", ["uid"], name: "index_oauth_applications_on_uid", unique: true, using: :btree
 
+  create_table "order_lines", force: true do |t|
+    t.integer  "unit",       default: 1, null: false
+    t.integer  "price",                  null: false
+    t.integer  "quantity",               null: false
+    t.integer  "order_id"
+    t.integer  "product_id"
```
Comment (enricostano) [inline db/schema.rb]: `product_id` and `order_id` should be `null: false` right?

File: db/schema.rb [hunk 3]
```diff
@@ -73,6 +73,29 @@
...
+    t.datetime "created_at"
+    t.datetime "updated_at"
+  end
+
+  add_index "order_lines", ["order_id"], name: "index_order_lines_on_order_id", using: :btree
+  add_index "order_lines", ["product_id"], name: "index_order_lines_on_product_id", using: :btree
+
+  create_table "orders", force: true do |t|
+    t.integer  "user_id"
+    t.integer  "group_id"
```
Comment (enricostano) [inline db/schema.rb]: `user_id` and `group_id` should be `null: false` right?

File: db/schema.rb [hunk 4]
```diff
@@ -99,10 +122,20 @@
     t.datetime "updated_at"
   end
 
+  create_table "suppliers", force: true do |t|
+    t.integer  "group_id"
+    t.integer  "producer_id"
```
Comment (enricostano) [inline db/schema.rb]: `group_id` and `producer_id` should be `null: false` right?

File: db/schema.rb (hunk 2 above)
Comment (sauloperez) [inline db/schema.rb]: yep

**Comment by sauloperez** [comment]
I thought I have added these nulls... might be lost in some of my computers :trollface:

**Comment by sauloperez** [comment]
Take a look again. Now it should be all set.

**Comment by enricostano** [comment]
Please add some seed data and we're ready to merge and work on CRUD

Many thanks @sauloperez :clap:

**Pull request merged**